### DISCUSSION TITLE
Training ParallelWaveGAN on an Arabic dataset

### DISCUSSION THREAD

**Discussion by ahgarawani** [coqui-ai/TTS#565] [General Q&A] [answered]
Hi, I would like to train the ParallelWaveGAN model on an Arabic dataset but I don't know where to begin, Honestly the repos is quite confusing to navigate, can anybody help?

**Comment by Sadam1195** [comment]
As far as I understand if you want to train `ParallelWaveGAN` you need to use  https://github.com/coqui-ai/TTS/blob/main/TTS/bin/train_vocoder_gan.py and use the config here https://github.com/coqui-ai/TTS/blob/main/TTS/config/shared_configs.py so the commands would look something like this.
`python train_vocoder_gan.py -config_path here -output_path here -dataset_path here` 
But @erogol could tell better.

**Reply by ahgarawani** [reply]
Thank you. It seems like I need to specify which GAN model I want to use, how do I do that ?

**Reply by Sadam1195** [reply]
I am not sure if the code in current version is complete or else I am not able to figure out without much of the documentation. If you really want to train `ParallelWaveGAN` you can use previous versions of the repo and train it.

**Comment by ahgarawani** [comment]
@erogol I would appreciate it if you can spare the time and point me in the right direction. Thanks.

**Reply by Sadam1195** [reply]
hey you can use the new trainer API and create a .py file for `ParallelWaveGAN` in the same manner like presented in here https://github.com/coqui-ai/TTS/tree/trainer-api/recipes/ljspeech for other vocoders.

checkout git repo with trainer-api branch

@ahgarawani

**Reply by Sadam1195** [reply]
create a file train_ParallelWaveGAN.py with following code

```
import os
from TTS.vocoder.configs import  ParallelWaveganConfig
from TTS.trainer import init_training, Trainer, TrainingArgs

output_path = os.path.dirname(os.path.abspath(__file__))
config =  ParallelWaveganConfig(    
batch_size=32,    
eval_batch_size=16,    
num_loader_workers=4,    
num_eval_loader_workers=4,    
run_eval=True,    
test_delay_epochs=-1,    
epochs=1000,    
seq_len=8192,   
pad_short=2000,   
use_noise_augment=True,    
eval_split_size=10,    
print_step=25,    
print_eval=True,    
mixed_precision=False,    
lr_gen=1e-4,    
lr_disc=1e-4,    
data_path=os.path.join(output_path, "../LJSpeech-1.1/wavs/"),    
output_path=output_path,
)
args, config, output_path, _, c_logger, tb_logger = init_training(TrainingArgs(), config)
trainer = Trainer(args, config, output_path, c_logger, tb_logger)trainer.fit()
```

and then call `python train_ParallelWaveGAN.py`

@ahgarawani

**Reply by Sadam1195** [reply]
I am not sure if this is required but if code breaks follow this last step

Once you have done that create a `ParallelWaveGAN.json` 
by copying fields from 
https://github.com/coqui-ai/TTS/blob/14039946984f8f1b9fe0e7e8e101f78ab0b980eb/TTS/vocoder/configs/parallel_wavegan_config.py 
in the same format as mentioned here https://github.com/coqui-ai/TTS/blob/trainer-api/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json 

https://github.com/coqui-ai/TTS/tree/trainer-api/recipes/ljspeech

```
CUDA_VISIBLE_DEVICES="0" python TTS/recipes/ljspeech/train_ParallelWaveGAN.py 
--config_path ./ParallelWaveGAN.json \                                                    
 --coqpit.output_path    ./output_path     \                                       
 --coqpit.datasets.0.path ./LJSpeech-1.1/    \                                                     
 --coqpit.audio.stats_path ./scale_stats.npy \
```

**Comment by erogol** [comment, marked as answer]
The easiest way is to use this new trainer API branch https://github.com/coqui-ai/TTS/tree/trainer-api

Go to the LJSpeech recipes and take the model you want edit the data field in the config and run your model. I also suggest you double-check the configuration fields if they are as you expected. Config fields are documented by the docsting.

**Comment by ahgarawani** [comment]
Thank you both very much
